packages/issuer/lib/VcIssuer.ts

```diff
@@ -24,31 +27,37 @@ import {
   TYP_ERROR,
 } from '@sphereon/openid4vci-common'
 import { ICredential, W3CVerifiableCredential } from '@sphereon/ssi-types'
+import { v4 } from 'uuid'
 
 export class VcIssuer {
   _issuerMetadata: IssuerMetadata
   _userPinRequired?: boolean
   _issuerCallback?: CredentialIssuerCallback
   _verifyCallback?: JWTVerifyCallback
-  private readonly _stateManager: ICredentialOfferStateManager
+  private readonly _stateManager: IStateManager<CredentialOfferState>
+  private readonly _cNonceStateManager: IStateManager<CNonceState>
+  // TODO add config option
+  private readonly _cNonceExpiresIn: number = parseInt(process.env.C_NONCE_EXPIRES_IN as string) * 1000 || 90 * 1000
 
   constructor(
     issuerMetadata: IssuerMetadata,
     args: {
       userPinRequired?: boolean
-      stateManager: ICredentialOfferStateManager
+      stateManager: IStateManager<CredentialOfferState>
+      cNonceStateManager: IStateManager<CNonceState>
       callback?: CredentialIssuerCallback
       verifyCallback?: JWTVerifyCallback
     }
   ) {
     this._issuerMetadata = issuerMetadata
     this._stateManager = args.stateManager
+    this._cNonceStateManager = args.cNonceStateManager
```

Comment:
Just call that nonceManager. It becomes very confusing is you call one thing a state manager and then another a cnoncestate manager

Reply:
Fixed!